packages/ai/src/Button.ts

```diff
@@ -121,6 +143,35 @@ class Button extends UI5Element {
 	@property({ type: Boolean })
 	fadeIn = false;
 
+	/**
+	 * Flag for transition between button without end icon or arrow button
+	 * and button with end icon or arrow button.
+	 * This property is animation related only.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	buttonToMenu? = false;
+
+	/**
+	 * Flag for transition between button with end icon or arrow button
+	 * and button without end icon or arrow button.
+	 * This property is animation related only.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	menuToButton? = false;
```

Comment:
There are no advantages to add a classes because we should have flag properties AND to apply classes. Now it is done automatically only by flags/attributes.